pkg/scaling/resolver/scale_resolvers.go

```diff
@@ -191,6 +191,10 @@ func ResolveAuthRefAndPodIdentity(ctx context.Context, client client.Client, log
 			authParams["awsRoleArn"] = serviceAccount.Annotations[kedav1alpha1.PodIdentityAnnotationEKS]
 		} else if podIdentity.Provider == kedav1alpha1.PodIdentityProviderAwsKiam {
 			authParams["awsRoleArn"] = podTemplateSpec.ObjectMeta.Annotations[kedav1alpha1.PodIdentityAnnotationKiam]
+		} else if podIdentity.Provider == kedav1alpha1.PodIdentityProviderAzure {
+			if podIdentity.GetIdentityID() == "" {
+				logger.Info("WARNING: IdentityID of PodIdentity is empty or nil")
+			}
```

Comment:
Is it a valid condition if the default is `nil`? I'd say no because it means they are passing an empty value which is probably not intended

Reply:
The problem is that `GetIdentityID()` returns "" if it's `nil`

```golang
func (a *AuthPodIdentity) GetIdentityID() string {
    if a.IdentityID == nil {
        return ""
    } else {
        return *a.IdentityID
    }
}
```

Reply:
@tomkerkhove @JorTurFer I add a webhook to check if identity id is nil or empty. So let's assume that  GetIdentityID will be valid if the request pass the webhook. So is it necessary to hint user that podidentity is using the default value? Or just printing a regular log is enough?

Reply:
The long goal of the change is  to block identityId when it is an empty string because we had end-users where tokenization failed and accidentally used an empty string - So let's block those.

However, if I'm not mistaken the current default value was an empty string, so as part of this change we need to use `nil` as default instead of `""` to prevent end-users from being blocked when they did not specify the property

Reply:
Is it OK to keep the warning? Or should I modify more? @tomkerkhove @JorTurFer

Reply:
If the resource is successfully blocked then I'm not sure if we need to cause panic by logging a warning. Does that make sense?

Reply:
I updated the log and removed the warning.